README.md

```diff
@@ -127,40 +129,100 @@ $sslcert = @("EOT")
 -----END CERTIFICATE-----
 |-EOT
 
-$dbpass = Sensitive(Deferred(conjur::secret, ['production/postgres/password', {
+$dbpass = Deferred(conjur::secret, ['production/postgres/password', {
   appliance_url => "https://my.conjur.org",
   account => "myaccount",
   authn_login => "host/myhost",
   authn_api_key => Sensitive("2z9mndg1950gcx1mcrs6w18bwnp028dqkmc34vj8gh2p500ny1qk8n"),
   ssl_certificate => $sslcert
-}]))
+}])
 ```
 
-#### Sensitive data type
+#### Example usage
 
-Note `conjur::secret` returns values wrapped in a `Sensitive` data type. In
-some contexts, such as string interpolation, it might cause surprising results
-(interpolating to `Sensitive [value redacted]`). This is intentional, as it
-makes it more difficult to accidentally mishandle secrets.
+```puppet
+node 'server-123' {
+  $db_password = Deferred(conjur::secret, ['inventory/db-password'])
+
+  file { '/tmp/creds.txt':
+    ensure => file,
+    mode => '0600',
+    content => $db_password,
+  }
+
+  file { '/tmp/creds.ini':
+    ensure => file,
+    mode => '0600',
+    content => Deferred('inline_epp', [
```

Comment:
what is this?

Reply:
An example of using a conjur-retrieved variable in a templated file. Our primary example before was inadequate for almost all but the basic uses.

Reply:
can we add a comment above these two examples that say what they are?
"Example of writing a secret to a file"
"Example of using a secret in a templated file"